platforms/core-execution/build-cache-http/src/main/java/org/gradle/caching/http/HttpBuildCache.java

```diff
@@ -93,6 +96,7 @@ public void setUrl(@Nullable URI url) {
     /**
      * Returns the credentials used to access the HTTP cache backend.
      */
+    @ToBeKeptEagerProperty(because = "Read-only nested property")
```

Comment:
🤔 Should we mark these as `@Nested` instead?

Reply:
Done